spark/src/main/java/org/apache/iceberg/spark/SparkDataFile.java

```diff
@@ -91,14 +92,19 @@ public SparkDataFile wrap(Row row) {
     return this;
   }
 
+  public SparkDataFile withSpecId(int newSpecId) {
```

Comment:
Did you file the proposal to add specID into the datafile? I know i've had a few temporary prs where I thought about doing this. We could change the reader itself so that this is always populated and I think there is an active PR to do this somewhere as well.